bigframes/dataframe.py

```diff
@@ -2033,8 +2034,23 @@ def dropna(
 
         axis_n = utils.get_axis_number(axis)
 
+        if subset is not None and axis_n != 0:
+            raise NotImplementedError(
+                f"subset only supported when axis=0. {constants.FEEDBACK_LINK}"
+            )
+
         if axis_n == 0:
-            result = block_ops.dropna(self._block, self._block.value_columns, how=how)  # type: ignore
+            # subset needs to be converted into column IDs, not column labels.
+            if subset is None:
+                subset_ids = None
+            elif not utils.is_list_like(subset):
+                subset_ids = [self._block.label_to_col_id[subset]]
```

Comment:
The result of this seems to be a list of tuples, and don't work.

Reply:
This is supposed to cover the single column string case. A loop would give one character per iteration. I'll take a look at the error.

Reply:
In single column case the subset_ids seems looks like [(col_id,)], I assume one loop is needed for the tuple?

Reply:
From the defination ```def label_to_col_id(self) -> typing.Mapping[Label, typing.Sequence[str]]```, I think self._block.label_to_col_id[subset] will be a sequence(tuple)?  So I think a loop like for ```id in self._block.label_to_col_id[label]``` should be included?

Reply:
Chatted offline. `self._block.label_to_col_id[subset]` is a tuple. I've corrected.